controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -709,6 +709,7 @@ func (r *BareMetalHostReconciler) actionProvisioning(prov provisioner.Provisione
 		return actionContinue{}
 	}
 
+	info.host.Status.ProvisionCount++
```

Comment:
From the description, looks like the counter should be increased only in case of successfull provisioning (ie near `actionComplete{}`)